sklearnex/linear_model/linear.py

```diff
@@ -163,8 +163,11 @@ def _onedal_fit_supported(self, method_name, *data):
         n_samples = _num_samples(X)
         n_features = _num_features(X, fallback_1d=True)
 
-        # Check if equations are well defined
+        # Note: support for some variants was either introduced in oneDAL 2025.1,
+        # or had bugs in some uncommon cases in older versions.
         is_underdetermined = n_samples < (n_features + int(self.fit_intercept))
+        is_multi_output = hasattr(y, "shape") and len(y.shape) > 1 and y.shape[1] > 1

```

Comment:
https://github.com/intel/scikit-learn-intelex/blob/903fb2f97999d6e4acee02c771311a275412294a/onedal/utils/validation.py#L360  Try to use onedal's _num_features, it may be more robust.

Reply:
Looks like it doesn't work with lists:
```
TypeError: Unable to find the number of features from X of type list
```

Reply:
Could you try instead with sklearn's _num_features instead?

Reply:
Same error.

Reply:
Could you provide the list data which is causing _num_features to TypeError? 
I assume you are doing
```python
is_multi_output = _num_features(y) > 1
```

Reply:
Example where it errors passing a list:
```python
from sklearn.utils.validation import _num_features
_num_features([1,2,3])
```
```
TypeError: Unable to find the number of features from X of type list
```

Reply:
Use onedal's with _num_features(X, True) (has a fallback for 1d in the onedal implementation)

Reply:
Changed to use oneDAL's `_num_features`.

Reply:
Turns out `_num_features` behaves differently for different input types, so it's not a good replacement:
```python
import numpy as np
from onedal.utils import _num_features
_num_features([1,2,3], True) # =1
_num_features(np.array([1,2,3]), True) # =3
```